pipeline-runner/R/gem2s-6-prepare_experiment.R

```diff
@@ -1,221 +1,174 @@
-#  - Merging the samples for the current experiment
-#  - Adding metadata: cellsId, color_pool and gene annotation
-#  - Preparing dataProcessing json file
-
-
-prepare_experiment <- function(input, pipeline_config) {
-  message("Loading configuration...")
-  config <- RJSONIO::fromJSON("/input/meta.json")
-
-  print_config(6, "Prepare experiment", input, pipeline_config, config)
-
-  # Check which samples have been selected. Otherwiser we are going to use all of them.
-  if (length(config$samples) > 0) {
-    samples <- config$samples
-  } else {
-    samples <- gsub("\\..*", "", list.files("/output/rds_samples"))
-  }
-
-  message("Reloading samples rds for current experiment...")
-  scdata_list <- list()
-  for (sample in samples) {
-    scdata_list[[sample]] <- readRDS(paste("/output/rds_samples/", sample, ".rds", sep = ""))
-  }
-
-  # Merging samples and adding a prefix with the sample name. In pipeline we grep in barcodes to filter by sample.
+#' Prepare experiment for upload to AWS
+#'
+#'  1) Merges the samples for the current experiment
+#'  2) Adds metadata: cellsId, color_pool, and gene annotation
+#'  3) Preparing QC configuration
+#'
+#' @inheritParams download_cellranger
+#' @param prev_out  'output' slot from call to \code{create_seurat}
+#'
+#' @return prev_out \code{prev_out} with added slots 'scdata' containing merged
+#'   \code{SeuratObject} and 'qc_config' containing default config for QC steps.
+#'
```

Comment:
why don't you split each of those 3 items into separate functions? It will definitely make the code easier to read.

Reply:
Will do